crates/holochain_deterministic_integrity/src/lib.rs

```diff
@@ -1,4 +1,43 @@
-//! The Holochain Deterministic Integrity
+//! Holochain Deterministic Integrity (HDI) is Holochain's data model and integrity toolset for
+//! writing zomes.
+//!
+//! The logic of a Holochain DNA can be divided into two parts: integrity and coordination.
+//! Integrity is the part of the hApp that defines the data structure and validates data
+//! types and values. Coordination encompasses the domain logic and implements the functions
+//! to manipulate data.
+//!
+//! # Examples
+//!
+//! An example of an integrity zome with data definition and data validation can be found in the
+//! wasm workspace of the Holochain repository:
+//! <https://github.com/holochain/holochain/blob/develop/crates/test_utils/wasm/wasm_workspace/integrity_zome/src/lib.rs>.
+//!
+//! # Data definition
+//!
+//! The DNA's data model is defined in integrity zomes. They comprise all data type definitions
+//! as well as relationships between those types. Integrity zomes are purely definitions and do
+//! not contain functions to manipulate the data. Therefore a hApp's data model is encapsulated
+//! and completely independent of the domain logic, which is encoded in coordinator zomes.
+//!
+//! Taking a database as an analogy, an integrity zome is similar to the schema of a database.
+//! Coordinator zomes, in contrast, provide functionality to query and write to the database.
+//!
+//! Advantages of this approach are:
+//! * The DNA hash is constant as long as the integrity zomes remain the same. Changes to the
+//! domain logic, enclosed in coordinator zomes, do not affect the DNA hash. As a result, DNAs
+//! and therefore hApps can be constantly modified without creating a new peer network on every
```

Comment:
Is there a place we could link to that describes why changing the DNA hash creates a new peer network?

Reply:
I don't know a place, but briefly mentioned that changing a DNA hash does create a new peer network.